### DISCUSSION TITLE
A question about Optuna optimization for NN

### DISCUSSION THREAD

**Discussion by imrooki** [optuna/optuna#3936] [Q&A] [answered]
### Don't use GitHub Issues to ask support questions.

How can I reproduce this training process to obtain the same training and validation errors after obtaining hyperparameters using optuna tuning? How are the initialized weights during optuna training and the initial weights during replication guaranteed to be the same? Both ways, I've fixed the same random seeds, but don't get the same results. I hope to get an answer, thanks!

**Comment by nzw0301** [comment, marked as answer]
I suppose the objective function still has some stochastic behaviour or is affected by randomness in optuna, see https://scikit-learn.org/stable/common_pitfalls.html#controlling-randomness that is still a relevant resource to understand randomness. Since we cannot see the code, I'm afraid that I could not answer your question precisely.

**Reply by imrooki** [reply]
First of all, thank you for your reply. I think I only control torch.manual_ Seed (seed) without controlling random.seed (seed) and np.random Seed (seed). Of course, my original intention is to train the network built by this super parameter through the super parameter obtained by optuna. Unfortunately, even the same super parameter has not been tested and verified by optuna during training. I don't know if it is the reason I mentioned above.
